core/src/test/scala/org/apache/spark/sql/delta/DeltaAutoOptimizeSuite.scala

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.delta.sources.DeltaSQLConf
+import org.apache.spark.sql.delta.test.DeltaSQLCommandTest
+import org.apache.spark.sql.functions._
+
+class DeltaAutoOptimizeSuite extends QueryTest with SharedSparkSession with DeltaSQLCommandTest {
+  import testImplicits._
+
+  def writeDataToCheckAutoCompact(
+      numFiles: Int,
+      dataPath: String,
+      partitioned: Boolean = false,
+      mode: String = "overwrite"): Unit = {
+    val df = spark
+      .range(50000)
+      .map { _ =>
```

Comment:
These `map` followed by `toDF` could be replaced with a more Spark SQL bit:

```
.withColumn("colA", rand() * 10000000 cast "long")
.withColumn("colB", rand() * 1000000000 cast "int")
.withColumn("colC", rand() * 2 cast "int")
```

WDYT?